crates/ark/src/treesitter.rs

```diff
@@ -571,3 +571,22 @@ pub(crate) fn node_find_containing_call<'tree>(node: Node<'tree>) -> Option<Node
 
     None
 }
+
+pub(crate) fn point_end_of_previous_row(
```

Comment:
_Maybe_ a test or two for this? Particularly around row 0 behavior and newline behavior